hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/QueryUtil.java

```diff
@@ -217,25 +224,36 @@ public void setHazelcastInstance(HazelcastInstance hazelcastInstance) {
             this.hzInstance = hazelcastInstance;
         }
 
+        @Override
+        public void setNode(Node node) {
+            SecurityContext securityContext = node.securityContext;
+            if (securityContext != null && subject != null) {
+                this.ssc = securityContext.createSqlContext(subject);
+            }
+        }
+
         @Override
         public void setSerializationService(SerializationService serializationService) {
-            this.evalContext = new ExpressionEvalContextImpl(
-                    arguments,
-                    (InternalSerializationService) serializationService,
-                    Util.getNodeEngine(hzInstance));
+            initContext((InternalSerializationService) serializationService);
+        }
+
+        private void initContext(InternalSerializationService iss) {
+            this.evalContext = ExpressionEvalContext.createContext(arguments, hzInstance, iss, ssc);
```

Comment:
this depends on the order of invocations of `NodeAware` and `SerializationServiceAware`. I am not sure if it is guaranteed (is that documented anywhere?) but possibly we can assume that it will not change (maybe with assertion?).

Reply:
It's called by `HazelcastManagedContext#initialize`, and the assertion makes sense, agree.

Reply:
you used assertions that have a runtime cost and are needlessly complex and strict. you can simply assert if something is null or not or if the value changes. eg. multiple invocations of setHazelcastInstance are fine as long as it is invoked with the same value.

Reply:
#25631